### DISCUSSION TITLE
What approach would be the best for creating permanent "coding" notes ?

### DISCUSSION THREAD

**Discussion by mburazin** [BryanJenksCommunity/FAQ#28] [Q&A] [answered]
Hi Bryan, 

First of all, thanks for all the great content related to Obsidian. I love it!

I'm a web developer, and I want to build up my knowledge base in Obsidian using the Zettelkasten principles. I'm a beginner in Obsidian/Zettelkasten workflow so I appreciate if you could provide some of your insight into the following issue.

From what I understand, ideally, the permanent notes should be "atomic" and as independent as possible, so that they are focused, relatively short and reusable in different linked contexts. Makes sense.
It's relatively easy for me to apply this principle when I'm adding notes related to non-technical content.

But, when trying to do the same for concepts related to coding (web development, programming languages, etc.), often I find it very hard to have short notes. Usually my notes end up being quite long in order to explain a single concept.
Whenever I try to think in a direction of how I would split a permanent note which explains a relatively complex coding concept, I notice that it's very difficult to decouple it, and I get quite overwhelmed. 
Even if I would somehow manage to split a single concept into more notes, I would need to use the same code example in multiple ones, which would possibly make them non-atomic and highly coupled to each other.

Have you experienced similar pain-points in this area? Do you have any suggestions or insights? Or is it just me who's making things complicated here...?

Thanks!

//Marko.

**Comment by tallguyjenks** [comment, marked as answer]
For content that is less of "my own ideas" in a Zettelkasten and more factual and "technical" content, i usually don't go for atomicity, i instead favor an example that easily explains the concept to my future self in a way i will understand it best, with all necessary code and explanations. My notes for this may run long but that is where i leverage the different levels of headings, and using the document outline pane, a great example is my note on [C++ classes](https://publish.obsidian.md/bryan-jenks/C%2B%2B+Classes) the note is long but it contains a lot of the examples that I'll remember, but i also have other facets of the class code on the MOC note for that language/topic as you can see under the class section of my [C++ MOC](https://publish.obsidian.md/bryan-jenks/C%2B%2B#Classes+and+Object+Oriented+Programming). 

There's no one-size-fits-all approach to this, but this is just how I'm approaching it. I like notes liberally to others that may be related for might help in my future understanding when i forget a concept or need a refresher :)

**Reply by mburazin** [reply]
Thanks @tallguyjenks ! I took a look at your example of C++ classes and MOC. I wasn't even aware of the document outline pane, so I immediately enabled the plugin in my Obsidian app. 

It's exactly the kind of info I was looking for and now I feel ok with being more liberal when taking these kind of notes.

Thanks for putting my mind to rest 😉 

Until next time 😆

**Reply by tallguyjenks** [reply]
Glad it helped :)

**Reply by BindingOx** [reply]
Sorry to highjack the thread but your links to the notes here are showing a 404 error

**Reply by tallguyjenks** [reply]
@Macknificent yeah, the passage of time does end up in broken links. you can still search for the notes either on my public zettelkasten or devlog, my website has links to both in the navigation bar: https://www.bryanjenks.dev
